07_boolean_2/main.go

```diff
@@ -15,56 +15,343 @@ E.g.:
+		return false
+	}
+
+	if grade4 < limit {
+		return false
+	}
+
+	if grade5 < limit {
+		return false
+	}
+
+	return true
+}
+
+// check if grade avg >= limit
+func gradeAvgOver(grade1 int, grade2 int, grade3 int, grade4 int, grade5 int, limit int) bool {
+	var avg int = (grade1 + grade2 + grade3 + grade4 + grade5) / 5
+	return avg >= limit
+}
+
+func calcGrade(grade1 int, grade2 int, grade3 int, grade4 int, grade5 int) string {
+	if !gradeValid(grade1, grade2, grade3, grade4, grade5) {
+		return "invalid grade"
+	}
+
+	if gradeAvgOver(grade1, grade2, grade3, grade4, grade5, 90) {
+		if allGradesOver(grade1, grade2, grade3, grade4, grade5, 80) {
+			return "A+"
+		}
+		if allGradesOver(grade1, grade2, grade3, grade4, grade5, 70) {
+			return "A"
+		}
+		if anyGradeUnder(grade1, grade2, grade3, grade4, grade5, 70) {
+			return "A-"
+		}
+	}
+
+	if gradeAvgOver(grade1, grade2, grade3, grade4, grade5, 80) {
+		if allGradesOver(grade1, grade2, grade3, grade4, grade5, 80) {
+			return "B+"
+		}
+		if allGradesOver(grade1, grade2, grade3, grade4, grade5, 70) {
+			return "B"
+		}
+		if anyGradeUnder(grade1, grade2, grade3, grade4, grade5, 60) {
+			return "B-"
+		}
+	}
+
+	if gradeAvgOver(grade1, grade2, grade3, grade4, grade5, 70) {
+		if allGradesOver(grade1, grade2, grade3, grade4, grade5, 70) {
+			return "C+"
+		}
+		if allGradesOver(grade1, grade2, grade3, grade4, grade5, 60) {
+			return "C"
+		}
+		if anyGradeUnder(grade1, grade2, grade3, grade4, grade5, 60) {
+			return "C-"
+		}
+	}
+
+	return "grade calc error"
+}
+
+// TODO: Write a function that takes 3 numbers, representing the sides of a
+// triangle, and returns the type of the triangle:
+// - If any of the sides is less than or equal to 0, return "invalid"
+// - If the sum of the lengths of the two shortest sides is less than or
+// equal to the length of the longest side, return "invalid"
+// - If all three sides are equal, return "equilateral"
+// - If two sides are equal, return "isosceles"
+// - If no sides are equal, return "scalene"
+
+// any side <= 0 == true
+func anySideLess(side1 int, side2 int, side3 int) bool {
+	if side1 <= 0 || side2 <= 0 || side3 <= 0 {
+		return true
+	}
+	return false
+}
+
+// get sum of 2 biggest sides for sumLength
+func sumTwoBiggestSides(side1 int, side2 int, side3 int) int {
+	// check if side1 is bigger than others
+	if side1 > side2 && side2 > side3 {
+		return side1 + side2
+	}
+
+	if side1 > side3 && side3 > side2 {
+		return side1 + side3
+	}
+
+	if side2 > side3 && side3 > side1 {
+		return side2 + side3
+	}
+	// Return statement needed, placeholder/error message added
+	return 1337
+}
+
+// get smallest side for sumLength
+func smallestSide(side1 int, side2 int, side3 int) int {
+	if side1 > side2 && side2 > side3 {
+		return side3
+	}
+
+	if side1 > side3 && side3 > side2 {
+		return side2
+	}
+
+	if side2 > side3 && side3 > side1 {
+		return side1
+	}
+	// Return statement needed, placeholder/error message added
+	return 1337
```

Comment:
this will never be reached, you can remove the last if and return directly the last possible side.

Reply:
Removed.